core/include/jiminy/core/stepper/AbstractStepper.tpp

```diff
@@ -0,0 +1,50 @@
+
+#include "pinocchio/algorithm/joint-configuration.hpp"
+
+namespace jiminy
+{
+    template<typename F1>
+    AbstractStepper<F1>::AbstractStepper(F1 & f,
+                                         std::vector<Robot const *> robots):
+        f_(f);
+        robots_(robots)
+    {
+        // Empty
+    }
+
+
+    template<typename F1>
+    std::vector<vectorN_t> AbstractStepper<F1>::addToPosition(
+        std::vector<vectorN_t> const & positionIn,
+        std::vector<vectorN_t> const& velocityIn)
+    {
+        assert(positionIn.size() == robots_.size())
+        assert(velocityIn.size() == robots_.size())
+
+        std::vector<vectorN_t> positionOut;
+        for (int i = 0; i < robots_.size(); i++)
+        {
+            vectorN_t position = positionIn.at(i);
```

Comment:
`[i]` instead of `at` to avoid checking size systematically (they are not equivalent, `at` is slower)